File: 数据结构与算法/算法/贪心/code/money.go
package main

import "fmt"

func main() {
	var RMB = []int{200, 100, 20, 10, 5, 1}
	x := 1024
	count := 0
	for _, v := range RMB {
		use := int(x / v)
		x = x % v
		count += use
		if use == 0 {
			continue
		}
		fmt.Printf("需要使用 %d 的面额 %d 张，", v, use)
		fmt.Printf("剩余的金额为: %d。\n", x)
		if x == 0 {
			break
		}
	}
	fmt.Printf("共需要支付 %d 张.\n", count)
}

File: 数据结构与算法/算法/贪心/code/Money.java

public class Money{

    final static int[] RMB = new int[]{200, 100, 20,10, 5, 1};

    public static void main(String[] args) {
        int x = 1024;
        int count = 0;
        for (int i = 0; i < RMB.length; i++) {
            int use = x / RMB[i];
            count += use;
            if (use == 0) {
                continue;
            }
            x = x % RMB[i];
            System.out.printf("需要使用面额为 %d 的钞票 %d 张，", RMB[i], use);
            System.out.printf("剩余需要支持的金额：%d 。\n", x);
            if (x == 0) {
                break;
            }
        }
        System.out.printf("总共需要支付 %d 张。\n", count);
    }
}